src/main/resources/static/js/common.js

```diff
@@ -104,6 +104,17 @@ function isStringDefined(value) {
     return isDefined;
 }
 
+function parseJSON(value) {
```

Comment:
What does this function do? What type of value? is it hte JsonObject.
Write a comment of what the function do or what it is used for.

Reply:
Any value. It's not a json object and not of a defined type. It checks if the value is a number or true/false, if not then it tries to parse the value into json. I'm not sure this requires a specific comment to understand to be honest. I could put the regex into variables but it should be quite easy to see what it does.

Reply:
The issue is that when I look at the code I see only this function.
and it says only "parseJson" and Json in my world is a object "{ ... , ... }" with key and values. It can be any value . It doesn't say anythong what true, false, number, is used for and is nothing that can be derived from JSON name or JSON definition. Or am I missing something.
This is the background to my previous comment.

Reply:
Then we are in agreement, hahaha. If you read the PR description I try to explain why there is a check. The built in json parser treats numbers and booleans as json... so the check is there to make sure they are not positively validated and parsed. I only want things inside brackets to be validated and parsed.

Reply:
ok,, but a developer that do look at the code later do not have the PR info. So the code must be explainable enough so developer understand the code.

Reply:
I can put that piece of code in a function called exlcudeNumbersAndBooleans or something but the problem is in JSON.parse method. Maybe put a link to the documentation of this function? I'm not fully sure myself what it allows or doesn't allow. It's a strange function though.

Reply:
that function name says more than parseJson, since parseJson says more like a funtion that parse a Json object as string and return a Json object.
But this is my opinion, if other think I am thinking wrong and it should be treated differently, than you can skip my comment.

Reply:
I tried to find a regex to validate the json instead but I couldn't find any good expressions for javascript. There was an extremely accurate one in pcre but it doesn't work in javascript.

Reply:
I changed to parsing logic now. Maybe it makes more sense now?

Reply:
ok.